src/data/move-attrs/stat-stage-change-attr.ts

```diff
@@ -99,22 +98,34 @@ export class StatStageChangeAttr extends ChanceBasedMoveEffectAttr {
       switch (stat) {
         case Stat.ATK:
           if (this.selfTarget) {
-            noEffect = !user.getMoveset().find((m) => m instanceof AttackMove && m.category === MoveCategory.PHYSICAL);
+            noEffect = !user.getMoveset().find((m) => {
+              const move = m.getMove();
+              return move.isAttackMove() && move.category === MoveCategory.PHYSICAL;
```

Comment:
```suggestion
              return move.category === MoveCategory.PHYSICAL;
```
The `instanceof` checks in this are redundant since the category enums effectively enforce the type